packages/server/src/api/controllers/datasource.ts

```diff
@@ -245,6 +247,15 @@ export async function update(ctx: UserCtx<any, UpdateDatasourceResponse>) {
     datasource: await sdk.datasources.removeSecretSingle(datasource),
   }
   builderSocket?.emitDatasourceUpdate(ctx, datasource)
+  // send table updates if they have occurred
+  if (datasource.entities) {
+    for (let table of Object.values(datasource.entities)) {
+      const oldTable = baseDatasource.entities?.[table.name]
+      if (!oldTable || !isEqual(oldTable, table)) {
+        builderSocket?.emitTableUpdate(ctx, table, { includeOriginator: true })
+      }
+    }
+  }
```

Comment:
Cool, I wasn't aware we ever made schema updates against datasources directly. Since this is checking for diffs it'll hopefully not send any erroneous or duplicate events.

Reply:
Its really only in regards to relationships, due to the way existing relationships are defined the table API can't really handle it today (requires updating multiple tables at the same time) we could probably build a table API specifically for this, but emitting the table events seems like a good fix for now.